Tests/FluentAssertions.Specs/Collections/GenericCollectionAssertionsSpecs.cs

```diff
@@ -1931,5 +1931,237 @@ public void When_inspectors_count_does_not_equal_asserting_collection_length_it_
+                element => element == 1 || element == 2 || element == 3,
+                element => element == 1 || element == 2,
+                element => element == 1);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_some_predicates_have_multiple_matching_elements_and_most_restricitve_predicates_are_first_Satisfy_should_not_throw()
+        {
+            // Arrange
+            var collection = new int[] { 1, 2, 3, 4 };
+
+            // Act
+            Action act = () => collection.Should().Satisfy(
+                element => element == 1,
+                element => element == 1 || element == 2,
+                element => element == 1 || element == 2 || element == 3,
+                element => element == 1 || element == 2 || element == 3 || element == 4);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_second_predicate_matches_first_and_last_element_and_solution_exists_Satisfy_should_not_throw()
+        {
+            // Arrange
+            var collection = new int[] { 1, 2, 3 };
+
+            // Act
+            Action act = () => collection.Should().Satisfy(
+                element => element == 1,
+                element => element == 1 || element == 3,
+                element => element == 2);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_Satisfy_fails_then_failure_message_must_contain_predicates_without_matching_elements_and_elements_without_matching_predicates()
+        {
+            // Arrange
+
+            // These variables are introduced to make sure that expected results coming from variables
+            // will be printed out in user-friendly manner in the failure message
+            var oneText = "one";
+            var twoText = "two";
+            var threeText = "three";
+            var fourText = "four";
+
+            IEnumerable<SomeClass> collection = new[]
+            {
+                new SomeClass { Text = oneText, Number = 1 },
+                new SomeClass { Text = twoText, Number = 3 },
+                new SomeClass { Text = threeText, Number = 3 },
+                new SomeClass { Text = fourText, Number = 4 },
+            };
+
+            // Act
+            Action act = () => collection.Should().Satisfy(
+                new Expression<Func<SomeClass, bool>>[]
+                {
+                    element => element.Text == fourText && element.Number == 4,
+                    element => element.Text == twoText && element.Number == 2,
+                },
+                because: "we want to test formatting ({0})",
+                becauseArgs: "args");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage(
+@"Expected collection to satisfy all predicates because we want to test formatting (args), but:
```

Comment:
🤔 The full verification of the output should be diverted to the `PredicateLambdaExpressionValueFormatterSpecs`

Reply:
I think this particular test is still valuable as a form of documentation - so we can see the full output. But I have simplified it a bit in the latest iteration and moved formatting-specific validation to a separate test class. Pls let me know if you'd like more scenarios to be tested